src/parser/parser.yy

```diff
@@ -2041,10 +2045,20 @@ void nebula::GraphParser::error(const nebula::GraphParser::location_type& loc,
     errmsg = os.str();
 }
 
-void ifOutOfRange(const int64_t input,
+// parameter input accept the INTEGER value
+// which filled as uint64_t
+// so the conversion is expected
+void ifOutOfPositiveRange(const int64_t input,
+                  const nebula::GraphParser::location_type& loc) {
+    if ((uint64_t)input > 9223372036854775807ULL) {
+        throw nebula::GraphParser::syntax_error(loc, "Out of range:");
+    }
+}
+
+void ifOutOfNegativeRange(const int64_t input,
```

Comment:
The scanner has guaranteed no more than 9223372036854775808. Negative Numbers do not need to be checked

Reply:
Hex/Oct don't guaranteed.

Reply:
So I plan to check hex/oct number length when lex.